okhttp/src/main/java/okhttp3/internal/ws/RealWebSocket.kt

```diff
@@ -129,13 +122,18 @@ class RealWebSocket(
     }
 
     this.key = ByteArray(16).apply { random.nextBytes(this) }.toByteString().base64()
-    this.writerRunnable = Runnable {
-      try {
-        while (writeOneFrame()) {
+    this.writerTask = object : Task("${taskQueue.owner} Writer") {
+      override fun runOnce(): Long {
+        try {
+          while (writeOneFrame()) {
```

Comment:
Why is the loop here rather than returning 0?

Reply:
Good question! There’s a short answer and a long answer.

The short answer is that returning 0 is correct and I should do that.

The long answer is that I want to add a feature to TaskRunner so it requires fewer context switches. Right now returning 0 will context switch to the coordinator thread and then context switch back to the task thread. I think I can make it skip these context switches but it’s a code change I need to make.

I’m going to change this to return 0 now ’cause that’s correct locally, and I’ve got it in my plan to eliminate context switches in TaskRunner.